### PR TITLE
Add Continuous Integration to AeroSharp

### PR DESCRIPTION
**Pull request by Yamithorn**
## Description

Create continuous integration (CI) automation for packaging the library and publishing to NuGet, publishing preview versions on push to "preview" tags, and publishing release versions on merge to main in GitHub.

## Type of Change

- [ ] Bug Fix
- [X] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/wayfair-incubator/AeroSharp/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: .github/workflows/publish_preview.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Build
+        run: dotnet build src --configuration Release --no-restore
+      - name: Set VERSION variable from tag
+        run: echo "VERSION=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
+      - name: Pack NuGet
+        run: dotnet pack src --no-build --configuration Release /p:Version=${VERSION} --output src
+      - name: Publish NuGet
+        run: dotnet nuget push src/*.nupkg --source https://api.nuget.org/v3/index.json --api-key ${NUGET_KEY} --skip-duplicate
+        env:
+          NUGET_KEY: ${{ secrets.PREVIEW_NUGET_API_KEY }}
```
Comment (wbaldoumas) [inline .github/workflows/publish_preview.yml:36]: It doesn't look like there is a `PREVIEW_NUGET_API_KEY` available in this repo. The API key for the "Publish Preview" environment is just `NUGET_API_KEY`.

**Review by wbaldoumas — COMMENTED** [review]
It doesn't look like any NuGet packaging metadata has been added to the [project file](https://github.com/Yamithorn/AeroSharp/blob/yamithorn_test_and_release_open_source_library/src/AeroSharp/AeroSharp.csproj#L1-L16) for this library. We will need to get that added in this PR before publishing either previews or releases.

File: .github/workflows/publish_preview.yml (hunk 1 above)
Comment (Yamithorn) [reply .github/workflows/publish_preview.yml:36]: Fixed!

File: src/AeroSharp/AeroSharp.csproj [hunk 2]
```diff
@@ -2,8 +2,29 @@
 
     <PropertyGroup>
         <TargetFramework>netstandard2.1</TargetFramework>
+        <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+        <Version>0.1.0</Version>
+        <Authors>Wayfair</Authors>
+        <Description>Wrapper around the .NET Aerospike client that provides a variety of methods for storing and retrieving data on Aerospike</Description>
```
Comment (wbaldoumas) [inline src/AeroSharp/AeroSharp.csproj:8]: Nit: "data on Aerospike" => "data in Aerospike"

File: src/AeroSharp/AeroSharp.csproj [hunk 3]
```diff
@@ -2,8 +2,29 @@
...
+        <PackageLicenseFile>LICENSE</PackageLicenseFile>
+        <PackageProjectUrl>https://github.com/wayfair-incubator/AeroSharp</PackageProjectUrl>
+        <RepositoryUrl>https://github.com/wayfair-incubator/AeroSharp</RepositoryUrl>
+        <RepositoryType>git</RepositoryType>
+        <PackageTags>aerospike, aerosharp, serialization, policy validation</PackageTags>
     </PropertyGroup>
 
+    <PropertyGroup>
+        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    </PropertyGroup>
```
Comment (wbaldoumas) [inline src/AeroSharp/AeroSharp.csproj:19]: Nit: this is unrelated to NuGet packaging and shouldn't be introduced in this PR. Treating build warnings as build errors is something we may want to introduce later, but lets leave it out for now.

**Review by wbaldoumas — APPROVED** [review]

**Pull request merged**